qtism/data/storage/xml/XmlDocument.php

```diff
@@ -566,22 +568,59 @@ protected function inferVersion(): string
     private function validateDocComponent(QtiComponent $docComponent): void
     {
         $branchRules = $docComponent->getComponentsByClassName('branchRule', true);
+        $outcomeDeclarationIds = array_map(
+            static fn (OutcomeDeclaration $outcomeDeclaration) => $outcomeDeclaration->getIdentifier(),
+            $docComponent->getComponentsByClassName('outcomeDeclaration')->getArrayCopy()
+        );
+
         if ($branchRules->count() === 0) {
             return;
         }
 
         $errors = [];
+
         foreach ($branchRules as $branchRule) {
             $target = $branchRule->getTarget();
+
             if (empty($target)) {
                 $errors[] = 'BranchRule is missing a target attribute';
                 continue;
             }
 
+            if (in_array($target, BranchRule::SPECIAL_TARGETS, true)) {
+                continue;
+            }
+
             $targetElement = $docComponent->getComponentByIdentifier($target);
-            if ($targetElement === null && !in_array($target, BranchRule::SPECIAL_TARGETS, true)) {
+
+            if ($targetElement === null) {
                 $errors[] = sprintf('BranchRule target "%s" does not exist in the document', $target);
             }
+
+            $parentIdentifier = $branchRule->getParentIdentifier();
+            $parentElement = $docComponent->getComponentByIdentifier($parentIdentifier);
```

Comment:
- [ ] We may cache it maybe? As we may be pointing to the same test part more times and we can avoid querying it multiple times. Wdyt?